src/Models/Services/TelemetryService/TelemetryService.types.ts

```diff
@@ -22,8 +28,8 @@ export type SeverityLevel =
     | 'Critical';
 
 export interface IBaseTelemetryParams {
-    name: string;
     customProperties?: CustomProperties;
```

Comment:
CustomProperties are defined as Record<string, any>, is it possible to define a more strict interface for it by adding what do we expect there?

Reply:
the custom properties are inherently generic and unknown. We don't want a strict interface there because every event can then provide it's own collection of properties it wants to capture. That is how we get the values that are important to that event. Besides, the telemetry service doesn't really care what is in that collection, it simply passes it through to the logger. If we want to require certain metadata for an event, it should be one of the params provided to the look like we do for ComponentName.